formats/json/commonMain/src/kotlinx/serialization/json/internal/JsonNamesMap.kt

```diff
@@ -63,3 +63,23 @@ internal fun SerialDescriptor.getJsonNameIndexOrThrow(json: Json, name: String):
         throw SerializationException("$serialName does not contain element with name '$name'")
     return index
 }
+
+@OptIn(ExperimentalSerializationApi::class)
+internal inline fun Json.shouldCoerceValue(
```

Comment:
nit, but maybe it's better to name it `tryCoerceValue`?

All the call-sites of this method pass such values of lambdas that the part input is consumed. This is not something expected from the method with `shouldDoFoo(): Boolean` signature

Reply:
DynamicDecoder and TreeDecoder do not consume anything, they support random access. But for Streaming parser this is true indeed.